src/classes/ADDR_Addresses_UTIL.cls

```diff
@@ -0,0 +1,726 @@
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org 
+* @date 2016 
+* @group Addresses
+* @group-content ../../ApexDocContent/Addresses.htm
+* @description Addresses utility methods.
+*/
+public with sharing class ADDR_Addresses_UTIL {
+    
+    /* @description map from Country name to Country Code */
+    private static Map<String, String> validCountriesByLabel = new Map<String, String>();
+
+    /* @description map from State name to State Code */
+    private static Map<String, String> validStatesByLabel = new Map<String, String>();
+
+    /* @description map from Country Code to Country Name */
+    private static Map<String, String> validCountriesByCode = new Map<String, String>();
+    
+    /*******************************************************************************************************
+    * @description returns whether the Address's mailing fields are null
+    * @param addr the Address
+    * @return Boolean. true if all of the Mailing fields are empty
+    ********************************************************************************************************/
+    public static Boolean isAddressEmpty(Address__c addr) {
+        if (addr == null)
+           return false;        
+        return (addr.MailingStreet__c == Null && addr.MailingStreet2__c == Null && addr.MailingCity__c == Null &&
+           addr.MailingState__c == Null && addr.MailingPostalCode__c == Null && addr.MailingCountry__c == Null);
+    }
+    
+    public static Boolean isContactAccAddrInfoSame(Contact contact, Account acc) {
+        if ((contact == null && acc != null) || (contact != null && acc == null)) 
+           return false;
+        
+        Boolean isChanged = 
+           !ADDR_Addresses_UTIL.equalsCaseSensitive(contact.MailingStreet, acc.BillingStreet) ||
+           !ADDR_Addresses_UTIL.equalsCaseSensitive(contact.MailingCity, acc.BillingCity) ||
+           !ADDR_Addresses_UTIL.equalsCaseSensitive(contact.MailingState, acc.BillingState) ||
+           !ADDR_Addresses_UTIL.equalsCaseSensitive(contact.MailingPostalCode, acc.BillingPostalCode) ||
+           !ADDR_Addresses_UTIL.equalsCaseSensitive(contact.MailingCountry, acc.BillingCountry);
+
+        if (!isChanged && ADDR_Addresses_UTIL.isStateCountryPicklistsEnabled) {
+            isChanged = 
+                !ADDR_Addresses_UTIL.equalsCaseSensitive(String.valueOf(contact.get('MailingStateCode')), String.valueOf(acc.get('MailingStateCode'))) || 
+                !ADDR_Addresses_UTIL.equalsCaseSensitive(String.valueOf(contact.get('MailingCountryCode')), String.valueOf(acc.get('MailingCountryCode')));
```

Comment:
should be acc.get('BillingStateCode') and in the next line acc.get('BillingCountryCode')